src/NLog/Targets/DatabaseTarget.cs

```diff
@@ -522,6 +524,82 @@ private void WriteEventToDatabase(LogEventInfo logEvent)
             }
         }
 
+        /// <summary>
+        /// Set database type <paramref name="dbType"/> on <paramref name="dbDataParameter"/>
+        /// </summary>
+        /// <param name="dbDataParameter">The parameter.</param>
+        /// <param name="dbType"></param>
+        internal void SetParamType(IDbDataParameter dbDataParameter, string dbType)
+        {
+            if (string.IsNullOrEmpty(dbType))
+                return;
+            var lastIndexOfDot = dbType.LastIndexOf('.');
+
+            // Get just the fully qualified name of the enum but not the value
+            var enumName = lastIndexOfDot > 0 ? 
+                dbType.Substring(0, lastIndexOfDot) : string.Empty;
+
+            object enumValueObject = null;
+            Type paramType = dbDataParameter.GetType();
+            if (!string.IsNullOrEmpty(enumName))
+            {
+                try
+                {
+                    // check first to see if this is DbType otherwise use reflection to get it from the database parameter's assembly
+                    var enumDatabaseType = enumName == typeof (DbType).FullName ? typeof(DbType) : paramType.Assembly.GetType(enumName);
+
+                    // get the enum value for the database type
+                    var enumValueString = dbType.Substring(lastIndexOfDot + 1,
+                        dbType.Length - lastIndexOfDot - 1);
+                    var typeField = enumDatabaseType != null ? enumDatabaseType.GetField(enumValueString) : null;
+                    enumValueObject = typeField != null ? typeField.GetRawConstantValue() : null;
+                }
+                catch (Exception exception)
+                {
+                    InternalLogger.Trace(exception, "DatabaseTarget: unable to use " + enumName + " to set database column type.");
```

Comment:
We have new exception handling pattern in NLog 4.3 Please check: https://github.com/NLog/NLog/pull/1143#issue-125708489

I think this is a `warn`, also you can use `params `, so
`"DatabaseTarget: unable to use {0}" to set database column type.", enumName `

thanks!